src/backend/replication/walsender.c

```diff
@@ -3818,6 +3864,12 @@ WalSndKeepaliveIfNecessary(void)
 {
 	TimestampTz ping_time;
 
+	/*
+	 * Send an archival status message, if necessary.
+	 */
+	if (wal_sender_archiving_status_interval > 0)
```

Comment:
Probably best to add the WAL archiving check here too:
```suggestion
	if (XLogArchivingActive() && wal_sender_archiving_status_interval > 0)
```

Maybe we should even add a macro to `xlog.h` below the `#define XLogArchiveCommandSet()` like so:
```code
#define XLogArchiveCommandSet() (XLogArchiveCommand[0] != '\0')
#define XLogArchivingStatusReportingActive() \
	(XLogArchivingActive() && wal_sender_archiving_status_interval > 0)
```
Then we can use that macro to prevent missing a condition.